pkg/controller/decomission.go

```diff
@@ -103,9 +104,11 @@ func (c *Controller) checkForPoolDecommission(ctx context.Context, key string, t
 			}
 			if !found {
 				poolNamesRemoved = append(poolNamesRemoved, pstatus.SSName)
-				tenant.Status.Pools = append(tenant.Status.Pools[:i], tenant.Status.Pools[i+1:]...)
+			} else {
+				poolStatus = append(poolStatus, *pstatus.DeepCopy())
```

Comment:
If the only problem was that the index was out-of-bounds, then I don't see a reason to use `DeepCopy()` here.

Reply:
Not sure why we don't need to deep-copy in the old situation, but it won't be called every second, so LGTM.